html/modules/custom/reliefweb_post_api/src/Plugin/reliefweb_post_api/ContentProcessor/Job.php

```diff
@@ -0,0 +1,121 @@
+      throw new ContentProcessorException('Unallowed image URL: ' . $data['image']['url']);
+    }
+
+    $file_pattern = $provider->getUrlPattern('file');
+    foreach ($data['file'] ?? [] as $file) {
+      if (!empty($file['url']) && !$this->validateUrl($file['url'], $file_pattern)) {
+        throw new ContentProcessorException('Unallowed file URL: ' . $file['url']);
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(array $data): ?ContentEntityInterface {
+    // Ensure the data is valid.
+    $this->validate($data);
+
+    $bundle = $this->getbundle();
+    $provider = $this->getProvider($data['provider'] ?? '');
+
+    // Generate the UUID corresponding to the document URL.
+    $uuid = $this->generateUuid($data['url']);
+
+    // Load or create a new node.
+    $node = $this->entityRepository->loadEntityByUuid('node', $uuid) ??
+            $this->entityTypeManager->getStorage('node')->create([
+              'uuid' => $uuid,
+              'type' => $bundle,
+              'langcode' => $this->getDefaultLangcode(),
+              'uid' => $provider->getUserId(),
+            ]);
+
+    // Verify the bundle if the entity already exists.
+    if (!$node->isNew() && $node->bundle() !== $bundle) {
+      throw new ContentProcessorException(strtr('Existing entity with the UUID @uuid is not a @bundle.', [
+        '@uuid' => $uuid,
+        '@bundle' => $bundle,
+      ]));
+    }
+
+    // Skip if the node was marked as refused.
+    if (!$node->isNew() && $node->getModerationStatus() === 'refused') {
+      throw new ContentProcessorException(strtr('Skipping processing: existing entity with the UUID @uuid is marked as refused.', [
+        '@uuid' => $uuid,
+        '@bundle' => $bundle,
+      ]));
+    }
+
+    // Set the mandatory fields.
+    $node->title = $this->sanitizeString($data['title']);
+
+    $this->setTermField($node, 'field_career_categories', 'career_category', $data['career_categories'] ?? []);
+    $node->set('field_city', $this->sanitizeString($data['city'] ?? ''));
+    $this->setDateField($node, 'field_job_closing_date', $data['closing_date']);
+    $this->setTermField($node, 'field_country', 'country', $data['country'] ?? []);
+    $this->setTextField($node, 'field_how_to_apply', $data['how_to_apply'], format: 'markdown');
+    $this->setTextField($node, 'body', $data['body'], format: 'markdown');
+    $this->setTermField($node, 'field_job_type', 'job_type', $data['job_type'] ?? []);
+    $this->setTermField($node, 'field_job_experience', 'job_experience', $data['job_experience'] ?? []);
+    $this->setTermField($node, 'field_source', 'source', $data['source'] ?? []);
+    $this->setTermField($node, 'field_theme', 'theme', $data['theme'] ?? []);
+
+    // Emails to notify when the document is published.
+    $emails = implode(',', $data['notify'] ?? $provider->getEmailsToNotify() ?? []);
+    $this->setField($node, 'field_notify', $emails ?: NULL);
```

Comment:
Note that there is currently no such field for jobs currenty (see other comment on job specs).